libbeat/docs/event-conventions.asciidoc

```diff
@@ -1,37 +1,47 @@
-=== Event conventions
-
-==== Naming conventions
-
-For the field names, the Beats follow the naming conventions below:
-
-- All fields must be lower case
-- Snake case for combining words
-- Group related fields in sub documents, which means using the . notation. Groups are mostly described by common prefixes.
-- Prevent namespace duplication. If connections appears in the namespace, it's not needed in the sub document
-- Do not use complex abbreviations. A list of standardised abbreviations can be found below.
-- Organise the documents from the general to the details, which allows namespacing. The type should always be last, like .pct.
-- If two fields are the same but with different units, remove the less granular one
+=== Naming Conventions
+
+When creating events, use the following conventions for field names and abbreviations.
+
+[[field-names]]
+==== Field Names
+
+Use the following naming conventions for field names:
+
+- All fields must be lower case.
+- Use snake case (underscores) for combining words.
+- Group related fields into subdocuments, which means using the dot (.) notation. Groups are mostly described by common prefixes. 
+//QUESTION: I am not sure what you mean by common prefixes here. Perhaps examples would help?
+- Prevent namespace duplication. If connections appears in the namespace, it's not needed in the sub document 
+//QUESTION: I don't understand what you mean here. Can you clarify or give an example?
+- Do not use complex abbreviations. See <<abbreviations>>.
+//QUESTION: What do we mean by complex abbreviations? Can we just tell people to use standard abbreviations and avoid using abbreviations that aren't commonly known?
+- Organise the documents from general to specific to allow for namespacing. The type, such as `.pct`, should always be last. For example, `system.core.user.pct`
+- If two fields are the same, but with different units, remove the less granular one.
+//QUESTION: What if we want to  provide a less granular view, too? It would be useful to provide an example here.
 - In case the value correlates with the name of a nested document, use value inside the document
-- Do not use . in the names
-- Use singular and plural properly for the fields. Example: sec_per_request vs open_requests
-- Use singular names for metricsets. It easier to read the event created: system.process.load = 0.3
-
-
-The goal is to have a similar experience across all metrics.
+//QUESTON: I don't understand what you mean here. Can you clarify?
+- Do not use dots (.) in names. 
+//QUESTION: Wouldn't it be clearer to say, "Do not use dots (.) in individual field names. Dots are reserved for grouping related fields into subdocuments." Is that why you have this restriction?
+- Use singular and plural names properly to reflect the field content. For example: `sec_per_request` vs `open_requests`. 
+//QUESTION: I wonder if we should remove this rule since we sort of violate it ourselves (especially with our  abbreviations).
```

Comment:
All these rules were created when I went through all the existing metricsets and tried to unify them. The above "rule" was introduced as some services use `request_per_sec` others `requests_per_sec` and a decision had to made which one to use. I don't think that common sense will apply here as the natural thing to me initially felt like keep what the service provides, but that leaded to inconsistency.